src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -320,17 +320,29 @@ public Task NewSolutionSnapshotAsync(Solution solution, CancellationToken cancel
             return Task.CompletedTask;
         }
 
-        private static bool AnalysisEnabled(TextDocument document)
+        private static bool AnalysisEnabled(TextDocument document, IDocumentTrackingService documentTrackingService)
         {
             if (document.Services.GetService<DocumentPropertiesService>()?.DiagnosticsLspClientName != null)
             {
                 // This is a generated Razor document, and they want diagnostics, so let's report it
                 return true;
             }
 
+            if (!document.SupportsDiagnostics())
+            {
+                return false;
+            }
+
             // change it to check active file (or visible files), not open files if active file tracking is enabled.
             // otherwise, use open file.
-            return document.IsOpen() && document.SupportsDiagnostics();
+            if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(document.Project) == BackgroundAnalysisScope.ActiveFile)
```

Comment:
So active file analysis scope is no longer a solution crawler concept for all incremental analyzers, but just something that diagnostic incremental analyzer respects?

Reply:
Yes. The same already applied to `OpenFilesAndProjects` and `FullSolution`.